src/extensions/environments/environments.extension.ts

```diff
@@ -57,6 +62,28 @@ export class Environments {
     return env;
   }
 
+  /**
+   * get an environment Descriptor.
+   */
+  getDescriptor(component: Component): { id: string; icon: string } {
+    const defaultIcon = `
+      <svg width="50" height="50" xmlns="http://www.w3.org/2000/svg">
+          <circle cx="25" cy="25" r="20"/>
+      </svg>`;
+    // TODO: @guy after fix core extension then take it from core extension
+    const extension = component.config.extensions.findExtension(Environments.id);
```

Comment:
This line along with
`    const envId = extension.config.env;`
Is already appear a few times. should be extracted to be dry.